### DISCUSSION TITLE
didDownloadUpdate:(SUAppcastItem *)item is not getting called

### DISCUSSION THREAD

**Discussion by prasann-1433** [sparkle-project/Sparkle#2374] [General]
updater:(SPUUpdater *)updater didDownloadUpdate:(SUAppcastItem *)item; 

this delegate is not getting called after download is completed. 

I am updating from 1.27 to 2.41 and have updated code and new edSignature but for some reason the delegate didDownloadUpdate is not getting called. I now for sure the app.zip is getting downloaded since i checked that in my charles as well i see this warning in console 

Warning: Downloader's expected content length (%llu) != Appcast item's length (%llu)"

rest all delegates like below are getting called except the didDownloadUpdate am i doing something wrong. 

- (void)updater:(SPUUpdater *)updater didFindValidUpdate:(SUAppcastItem *)item;
- (void)updater:(SPUUpdater *)updater willDownloadUpdate:(SUAppcastItem *)item withRequest:(NSMutableURLRequest *)request;

**Comment by prasann-1433** [comment]
never mind, looks like there was typo, when i did copied the delegate from .h file it worked. But some times i am seeing the log after re-start of app 

Error: Failed to gain authorization required to update target

even though my app is in desktop where its in writable mode.

**Reply by zorgiepoo** [reply]
there should be more logs along with that message either in your app or in the Installer XPC process (if your app is sandboxed). If it is sandboxed, this may indicate an incorrect configuration sandbox setup.
